### ISSUE TITLE
investigate badly encoded qs_localities.qs_loc names

### ISSUE THREAD

**Issue by sevko** [pelias/quattroshapes-patch#9]
We use the `qs_loc` property in both the admin-lookup (`pelias/admin-lookup`) and the Quattroshapes importer (`pelias/quattroshapes-pipeline`), but it has some badly encoded data as shown in pelias/pelias#82. The `qs_loc_alt` property looks like it has better alternative data, but *only* when `qs_loc` contains unicode characters, or, in Postgres terms, when `length(qs_loc) != octet_length(qs_loc)`. Otherwise, it appears to be worse.

**Comment by missinglink** [comment]
This might be related to the comments at around 16:00 on this video:
https://archive.org/details/sotmus2013-Nathaniel_V._Kelso_and_David_Blackman_-_Introducing_Quattroshapes_--_-68099836

**Comment by sevko** [comment]
The idea is to investigate the `qs_loc` and `qs_loc_alt` properties in the `qs_localities` shapefile, and determine when one is better than the other (as per my observations above, `qs_loc_alt` appears to be better *only* when `qs_loc` contains one or more unicode characters). Then, if it makes sense to, add a query to [clean_data.sql](https://github.com/pelias/quattroshapes-patch/blob/master/clean_data.sql) to replace `qs_loc` with `qs_loc_alt` when the right condition is met.

add a query to [clean_data.sql](https://github.com/pelias/quattroshapes-patch/blob/master/clean_data.sql) to set `qs_loc = qs_loc_alt

**Comment by dianashk** [comment]
Another example:

"Culture Espresso"

[search](http://pelias.mapzen.com/search?input=culture%20espresso)

top result:

``` javascript
{
  type: "Feature",
  properties: {
    id: "925134012",
    layer: "osmnode",
    name: "Espresso",
    alpha3: "CZE",
    admin0: "Czech Republic",
    admin1: "Karlovarský",
    locality: "MariÃ¡nskÃ© LÃ¡znÄ",
    category: [
      "food",
      "retail",
      "nightlife"
    ],
    address: { },
    text: "Espresso, MariÃ¡nskÃ© LÃ¡znÄ, Karlovarský"
  },
  geometry: {
    type: "Point",
    coordinates: [
      12.70403,
      49.9728
    ]
  }
}
```

**Comment by dianashk** [comment]
Implement end-to-end tests for the repo  as part of this effort.

**Comment by dianashk** [comment]
Data team will provide new set of Quattroshapes data with these issues resolved.

**Comment by missinglink** [comment]
There is a bit of confusion over separate and distinct encoding issues going on; which may be compounding.

The data team have accepted that there are some issues in the quattroshapes data _but_ it seems that **the vast majority of issues are due to an error somewhere in the Pelias pipeline** which either causes more, or exacerbates the issue further.

I ran the latest version of quattroshapes files through our `ogr2ogr` simplification script today and the encoding was fine, so that is not the issue.

I suspect the issue lies somewhere in https://github.com/pelias/quattroshapes-patch which produces very nasty strings like this:

``` bash
    "qs_a1": "Ð¥Ð°ÑÐºÑÐ²ÑÑÐºÐ°",
    "qs_a0": "Ð£ÐºÑÐ°ÑÐ½Ð°",
```

without looking in to it more I'd hazard a guess that it's due to using regular expressions with a library which doesn't fully support variable-length (8bit/16bit) string encoding (quite common); although it could be happening anywhere in the codebase.

until the issue is fixed in this repo we should avoid reporting upsteam data encoding issues unless we are sure they are an error at source.

**Comment by missinglink** [comment]
related: https://github.com/pelias/api/issues/224

**Comment by riordan** [comment]
To be resolved by pelias/pelias#144

**Comment by dianashk** [comment]
This has been fixed as part of the WOF milestone.

http://pelias.github.io/compare/#/v1/place%3Fids=openstreetmap:venue:node:925134012
